libsolidity/ast/AST.cpp

```diff
@@ -514,6 +565,26 @@ VariableDeclarationAnnotation& VariableDeclaration::annotation() const
 	return dynamic_cast<VariableDeclarationAnnotation&>(*m_annotation);
 }
 
+ModifierArea::ModifierArea(SourceLocation const &_location,
+						   std::vector<ASTPointer<ModifierInvocation>> const &_modifiers,
+						   ASTPointer<std::vector<ASTPointer<FunctionDefinition>>> const &_functions,
+						   ASTPointer<std::vector<ASTPointer<ModifierArea>>> const &_subAreas,
+						   const ASTPointer<ModifierArea> &_parent) :
+		ASTNode(_location),
+		m_declaredModifiers(_modifiers),
+		m_functions(_functions),
+		m_subAreas(_subAreas),
+		m_parent(_parent)
+{
+	std::vector<ASTPointer<ModifierInvocation>> completeVector = m_declaredModifiers;
+	if (m_parent)
+	{
+		std::vector<ASTPointer<ModifierInvocation>> parentModifiers = m_parent->totalModifiers();
+		completeVector.insert(completeVector.begin(), parentModifiers.begin(), parentModifiers.end());
```

Comment:
Same as above. Also: why do you need the local variable ``completeVector``? You could just start by setting ``m_totalModifiers = m_declaredModifiers;`` and then continue with ``m_totalModifiers = parentModifiers + m_totalModifiers;``. Even better: do it the other way round: if there is a parent first set ``m_totalModifiers = parentModifers;`` and then use ``m_totalModifiers += m_declaredModifiers;`` after the ``ìf``.